extensions-core/s3-extensions/src/main/java/io/druid/storage/s3/S3DataSegmentPusherConfig.java

```diff
@@ -39,6 +39,9 @@
   @JsonProperty
   @Min(0)
   private int maxListingLength = 1000;
+  // use s3n by default for backward compatibility
+  @JsonProperty
+  private boolean useS3aSchema = false;
```

Comment:
can we add this to documentation ?

Reply:
@dclim @b-slim

Reply:
We are using S3 as deep storage. The batch ingestion by default uses S3N uri - is this the configuration to set so that it uses S3A url format? If so where should this be set? In the deep storage config along with other properties? Is it druid.storage.useS3aSchema = true?

@b-slim

Reply:
@palanieppan-m you have to add this to hadoop config file like `core-site.xml`. If you don't have any other configurations it would look something like:
```
<configuration>
  <property>
    <name>fs.s3a.endpoint</name>
    <value>s3.eu-central-1.amazonaws.com</value>
  </property>

  <property>
    <name>fs.s3.impl</name>
    <value>org.apache.hadoop.fs.s3a.S3AFileSystem</value>
  </property>

  <property>
    <name>fs.s3n.impl</name>
    <value>org.apache.hadoop.fs.s3a.S3AFileSystem</value>
  </property>

  <property>
    <name>fs.s3a.access.key</name>
    <value>${S3_ACCESS_KEY}</value>
  </property>

  <property>
    <name>fs.s3a.secret.key</name>
    <value>${S3_SECRET_KEY}</value>
  </property>
</configuration>
```

Don't forget to change endpoint if you're using different region.

Next create another config file named `jets3t.properties` with the following:
```
s3service.s3-endpoint = s3.eu-central-1.amazonaws.com

storage-service.request-signature-version=AWS4-HMAC-SHA256
```

Again change region if needed.

And lastly all this thing will only work with patched Hadoop libraries. I used Hadoop 2.7.3 from HDP 2.5.3 distribution. It is patched with aws-java-sdk-*:1.10.6 and works perfectly with it. So I replaced hadoop and aws jars and only then it worked. For more info you can take a look at this discussion: https://github.com/druid-io/druid/issues/4977

Reply:
@moscowart, your proposal is workaround for the default behavior of Druid which uses S3N URL scheme. To get that to work, you have to configure Hadoop to support it.

In this PR though, it looks like behavior can be changed such that Druid always S3A URL scheme instead, provided you turn on a config. I was enquiring whether why understanding is correct and what's the config name to turn on this behavior.